deployments.go

```diff
@@ -78,5 +80,22 @@ func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 		return r, err
 	}
 
+	if s, ok := opts.Updates.(status.SubscribableStream); ok {
+		for update := range s.Subscribe() {
```

Comment:
I think, like you were saying before, this will cause concurrent deployments to lock up because we open a transaction [here](https://github.com/mhahn/empire/blob/f3afbd7e24e98c275d932581fc1c9afe31c6cb32/empire.go#L574).

In reality, we only need to make [deploymentsService.deploy](https://github.com/mhahn/empire/blob/f3afbd7e24e98c275d932581fc1c9afe31c6cb32/deployments.go#L21) automic, so we could move opening a transaction to [here](https://github.com/mhahn/empire/blob/f3afbd7e24e98c275d932581fc1c9afe31c6cb32/deployments.go#L71)